app/src/forms/email/emailService.js

```diff
@@ -12,20 +12,28 @@ const generateContexts = (type, configData, submission, referer) => {
   let userTypePath = '';
   if (type === 'sendStatusAssigned') {
     contextToVal = [configData.assignmentNotificationEmail];
-    userTypePath = '/view?s=';
+    userTypePath = 'form/view?s=';
+  } else if (type === 'sendSubmissionAssigned') {
+    contextToVal = [configData.assignmentNotificationEmail];
+    userTypePath = 'user/view?s=';
+  } else if (type === 'sendSubmissionUnassigned') {
+    contextToVal = [configData.assignmentNotificationEmail];
+    userTypePath = 'user/view?s=';
   } else if (type === 'sendSubmissionConfirmation' && configData.form.showSubmissionConfirmation) {
-    contextToVal = [configData.body.to];
-    userTypePath = '/success?s=';
+    contextToVal = configData.form.submissionReceivedEmails;
+    userTypePath = 'form/success?s=';
   } else if (type === 'sendSubmissionReceived' && configData.form.submissionReceivedEmails) {
     contextToVal = configData.form.submissionReceivedEmails;
-    userTypePath = '/view?s=';
+    userTypePath = 'form/view?s=';
   }
   return [{
     context: {
       confirmationNumber: submission.confirmationId,
       title: configData.title,
       messageLinkText: configData.messageLinkText,
-      messageLinkUrl: `${service._appUrl(referer)}/form${userTypePath}${submission.id}`,
+      messageLinkUrl: `${service._appUrl(referer)}/${userTypePath}${submission.id}`,
```

Comment:
It'd be much better to have the query param part `?s=` be embedded into `messageLinkUrl`'s string directly instead of repeating it in the `userTypePath` variable.

Reply:
removed the '?s=' from the userTypePath variable and put directly into the messageLinkUrl.